Modules/posixmodule.c

```diff
@@ -14335,6 +14335,22 @@ os_get_terminal_size_impl(PyObject *module, int fd)
 }
 #endif /* defined(TERMSIZE_USE_CONIO) || defined(TERMSIZE_USE_IOCTL) */
 
+/*[clinic input]
+os._get_cpu_count_config
+
+Private function to get PyConfig.cpu_count is overridden or not
```

Comment:
This doesn't belong in posixmodule, it is not platform specific.  `Python/sysmodule.c` perhaps?

and rather than an internal API returning two magic strings, how about an internal API that just returns the overridden value. Code calling it should only use it if it returns an int > 0, anything <= 0 means invalid/default/not-overridden.

That way os_cpu_count_impl below could just call the c impl function from sysmodule directly rather than having a copy of similar code.

Reply:
> and rather than an internal API returning two magic strings, how about an internal API that just returns the overridden value. Code calling it should only use it if it returns an int > 0, anything <= 0 means invalid/default/not-overridden.

Well, Victor and I worry about someone get the CPU count from the `_get_cpu_count_config`...

Reply:
FYI, -2 will be the process mode.

Reply:
I am worrying about this kinds of code, we never expect but somebody can do.

```python
cpu_count = sys._get_cpu_count_config() if sys._get_cpu_count_config() > 0else os.process_cpu_count()
```

Reply:
They could also `import posix` and do that from this location. I'm not worried about anyone using an `_` prefixed private undocumented api to shoot themselves in the foot.

Bikeshedding could come up with other more or less tempting names, but I don't think it matters too much. If it starts with `_` and we keep it undocumented, it isn't for public consumption.  `_get_overridden_cpu_count()` `_get_eaten_by_a_grue()` `_get_do_not_press_this_button_press_count()` ...

we've got plenty of other _ functions in this category in modules already.

Reply:
>  I'm not worried about anyone using an _ prefixed private undocumented api to shoot themselves in the foot.
> we've got plenty of other _ functions in this category in modules already.

Okay!